packages/plugin-core/src/views/LocalGraphView.ts

```diff
@@ -0,0 +1,178 @@
+import { AnalyticsUtils } from "../utils/analytics";
+import { VSCodeUtils } from "../vsCodeUtils";
+import { WebViewUtils } from "./utils";
+
+export class LocalGraphView implements vscode.WebviewViewProvider {
+  public static readonly viewType = DendronTreeViewKey.LOCAL_GRAPH_VIEW;
+  private _view?: vscode.WebviewView;
+  private _ext: IDendronExtension;
+  constructor(extension: IDendronExtension) {
+    this._ext = extension;
+    this._ext.context.subscriptions.push(
+      vscode.window.onDidChangeActiveTextEditor(this.onOpenTextDocument, this)
+    );
+  }
+
+  public postMessage(msg: DMessage) {
+    this._view?.webview.postMessage(msg);
+  }
+
+  public async resolveWebviewView(
+    webviewView: vscode.WebviewView,
+    _context: vscode.WebviewViewResolveContext<unknown>,
+    _token: vscode.CancellationToken
+  ) {
+    this._view = webviewView;
+    await WebViewUtils.prepareTreeView({
+      ext: this._ext,
+      key: DendronTreeViewKey.LOCAL_GRAPH_VIEW,
+      webviewView,
+    });
+    webviewView.webview.onDidReceiveMessage(
+      this.onDidReceiveMessageHandler,
+      this
+    );
+  }
+  async onDidReceiveMessageHandler(msg: GraphViewMessage) {
+    const ctx = "LocalGraphView:onDidReceiveMessage";
+    Logger.info({ ctx, data: msg });
+    switch (msg.type) {
+      case GraphViewMessageEnum.onSelect: {
+        const note = this._ext.getEngine().notes[msg.data.id];
+        await new GotoNoteCommand(this._ext).execute({
+          qs: note.fname,
+          vault: note.vault,
+        });
+        AnalyticsUtils.track(GraphEvents.LocalGraphSidePanel, {
+          message: GraphViewMessageEnum.onSelect,
+        });
+        break;
+      }
+      case GraphViewMessageEnum.onGetActiveEditor: {
+        const editor = VSCodeUtils.getActiveTextEditor();
+        this.onOpenTextDocument(editor);
+        break;
+      }
+      case DMessageEnum.MESSAGE_DISPATCHER_READY: {
+        // if ready, get current note
+        const note = this._ext.wsUtils.getActiveNote();
+        if (note) {
+          Logger.debug({
+            ctx,
+            msg: "got active note",
+            note: NoteUtils.toLogObj(note),
+          });
+        }
+        if (note) {
+          this.refresh(note);
+        }
+        break;
+      }
+      case GraphViewMessageEnum.onRequestGraphStyleAndTheme: {
+        // Set graph styles
+        const styles = GraphStyleService.getParsedStyles();
+        const graphTheme = MetadataService.instance().getGraphTheme();
+        if (this._view && (styles || graphTheme)) {
+          this._view.webview.postMessage({
+            type: GraphViewMessageEnum.onGraphStyleAndThemeLoad,
+            data: {
+              styles,
+              graphTheme,
+              isSidePanel: true,
+            },
+            source: "vscode",
+          });
+        }
+        break;
+      }
+
+      case GraphViewMessageEnum.onLocalGraphVisible: {
+        AnalyticsUtils.track(GraphEvents.LocalGraphSidePanel, {
```

Comment:
Rather than track 'visible' here, let's track it on the VSCode panel's visibility event.  `vscode.WebviewView.onDidChangeVisibility`